### DISCUSSION TITLE
Dream bigger?

### DISCUSSION THREAD

**Discussion by lewoudar** [exaloop/codon#264] [Ideas]
Hi folks,
I am really fascinated by this project. It makes me think about the [Crystal](https://crystal-lang.org/) programming language which started as a compiler for [Ruby](https://www.ruby-lang.org/en/). I have dreamed secretly to see a project like this for python and codon seems close.

And here is my point, just like Crystal folks decide to create a fully operational programming language, why not just do the same instead of trying to catch all the mysteries of Python and try to compile every python project with techniques like [codon-jit](https://pypi.org/project/codon-jit/). I think it will just create infinite work going this way.
Managing the project like a full programming language and giving the possibility to generate a python extension like you are doing on the [pyext](https://github.com/exaloop/codon/tree/pyext) branch seems more reasonable to me.

Also, I know that developing a project like this one means a lot of financial/human resources but I'm not sure your [license](https://mariadb.com/bsl11/) will help you in this case. It will be a barrier for small businesses that can be great advocates for your project. Maybe you should clearly ask for sponsorship? 😅 Honestly I'm not an expert on this part.

This was just my Saturday morning thinking 😆. Nevertheless, I'm happy with the project and hope it will go further 🙃

**Comment by learnforpractice** [comment]
Nice perspective! If there isn't too much insistence on compatibility, perhaps the saved energy could be used for other more pressing matters. Honestly, I don't think it's possible for Codon to maintain full compatibility with Python. The success of the Crystal programming language serves as an inspiring example.

**Comment by Naonao9527** [comment]
1

**Comment by inumanag** [comment]
Thank you for your input! Yeah, we are currently seriously considering this approach; will share more details maybe later on :)

**Reply by lewoudar** [reply]
Interesting :)
